src/plugins/intel_cpu/src/mkldnn_infer_request.cpp

```diff
@@ -502,146 +488,402 @@ void MKLDNNPlugin::MKLDNNInferRequest::SetBlob(const std::string& name, const In
+    }
+
+    const auto &outMap = graph->outputNodesMap;
+    auto output = outMap.find(name);
+    if (output != outMap.end()) {
+        if (_outputs.find(name) == _outputs.end()) {
+            auto outputNode = modelOutputsMap.find(name);
+            if (modelOutputsMap.find(name) != modelOutputsMap.end()) {
+                const auto shape = outputNode->second->get_input_partial_shape(0);
+                bool isDynamic = shape.is_dynamic();
+
+                if (!data) {
+                    InferenceEngine::SizeVector dims;
+                    if (isDynamic) {
+                        dims = InferenceEngine::SizeVector(shape.rank().get_length(), 0);
+                    } else {
+                        dims = shape.to_shape();
+                    }
+
+                    InferenceEngine::TensorDesc desc(InferenceEngine::details::convertPrecision(outputNode->second->get_input_element_type(0)),
+                                                     dims, InferenceEngine::TensorDesc::getLayoutByRank(dims.size()));
+
+                    data = make_blob_with_precision(desc);
+                    data->allocate();
+                } else {
+                    if (!shape.compatible(ov::PartialShape(data->getTensorDesc().getDims()))) {
+                        IE_THROW(ParameterMismatch) << "Network input and output use the same name: " << name << ", but expect blobs with different shapes.";
+                    }
+
+                    const auto netOutPrc = InferenceEngine::details::convertPrecision(outputNode->second->get_input_element_type(0));
+                    if (netOutPrc != data->getTensorDesc().getPrecision()) {
+                        IE_THROW(ParameterMismatch)
+                                    << "Network input and output use the same name: " << name << " but expect blobs with different precision: "
+                                    << data->getTensorDesc().getPrecision() << " for input and " << netOutPrc
+                                    << " for output.";
+                    }
+                }
+
+                _outputs[name] = data;
+                if (!isDynamic && !externalPtr.count(name) &&
+                    data->getTensorDesc() == MemoryDescUtils::convertToTensorDesc(output->second->getParentEdgesAtPort(0)[0]->getMemory().getDesc()) &&
+                        !graph->getProperty().batchLimit) {
+                    externalPtr[name] = data->buffer();
+                }
+            } else {
+                IE_THROW() << "Blob with name: " << name << " exists in MKLDNN graph, but absents in network outputs";
+            }
+        }
+        data = _outputs[name];
+    }
+
+    if (!data) {
+        IE_THROW() << "Cannot find blob with name: " << name;
+    }
+
+    return data;
 }
 
-void MKLDNNPlugin::MKLDNNInferRequest::ThrowIfCanceled() const {
-    if (_asyncRequest != nullptr) {
-        _asyncRequest->ThrowIfCanceled();
+void MKLDNNPlugin::MKLDNNInferRequestNewApi::PushInputData() {
```

Comment:
looks pretty the same with `MKLDNNInferRequestOldApi` version. Can we extract the common part?